include/syscall.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2017, Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache 2.0
+ */
+
+
+#ifndef _ZEPHYR_SYSCALL_H_
+#define _ZEPHYR_SYSCALL_H_
+
+/* Fixed system call IDs. We use #defines instead of enumeration so that if
+ * system calls are retired it does not shift the IDs of other system calls.
```

Comment:
Do we care if syscall IDs are recycled? If someone rebuilds an application with the retired syscall, it should use the new numbers anyway.

Reply:
There will come a time where the application and kernel can be built and loaded separately, we already have requests for it, and I don't want to impede that. My feeling is that this type of interaction between the app and kernel should be graven in stone.